portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,7 +28,17 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    ArrayList<String> greetings = new ArrayList<>();
+    greetings.add("Bonjour!");
+    greetings.add("Hola!");
+    greetings.add("Konichiwa!");
+    greetings.add("Hello!");
+    greetings.add("Ni Hao!");
+    
+    Gson gson = new Gson();
+    String json = gson.toJson(greetings);
+
     response.setContentType("text/html;");
```

Comment:
I believe you have to change this to `response.setContentType("application/json;");` it makes so JS automatically converts it into a JSON object. Did you test run this and it worked? Maybe it does something else.

Reply:
It worked without changing the content type to "application/json" (and no error pops up yet), probably because `json` is a string here. Will change the content type to "application/json" to be precise.